logistic-regression.py

```diff
@@ -0,0 +1,86 @@
+import time
+
+import numpy as np
+import perfplot
+
+from sklearn.datasets import make_classification, load_digits
+from sklearn.model_selection import train_test_split
+from sklearn import preprocessing
+
+from sklearn.linear_model import LogisticRegression
+from xlogistic import XP_LogisticRegression
+
+
+def SK_LogisticRegression(xp, n_features, n_classes):
+    # need to ignore the arguments for sklearn
+    return LogisticRegression(solver="lbfgs", C=1e5)
+
+
+class XP:
+    def __init__(self, xp, LR):
+        self.xp = __import__(xp)
+
+        # Helper to convert explicitly to NumPy for the minimization:
+        if xp == "numpy":
+            self.xp.get = lambda x: x.copy()
+        elif xp == "cupy":
+            self.xp.get = lambda x: x.get()
+        elif xp == "cunumeric":
+            self.xp.get = lambda x: x.__array__()
+
+        self.LR = LR
+
+    def setup(self, n=1_000_000):
+        xp = self.xp
+
+        X, y = make_classification(
+            n_classes=12, n_features=40, n_informative=32, n_samples=n, random_state=42
+        )
+
+        X = xp.array(X)
+        y = xp.array(y)
+
+        X_train, X_test, y_train, y_test = train_test_split(
+            X, y, test_size=0.15, random_state=42
+        )
+        X_train = xp.array(X_train)
+        X_test = xp.array(X_test)
+        y_train = xp.array(y_train)
+        y_test = xp.array(y_test)
+
+        print(f"{X_train.shape=}")
+
+        return X_train, X_test, y_train, y_test
+
+    def time(self, X_train, X_test, y_train, y_test):
+        n_features = X_train.shape[1]
+        n_classes = np.unique(y_train).shape[0]
+        lr_ = self.LR(self.xp, n_features, n_classes)
+        tic = time.perf_counter_ns()
+        lr_.fit(X_train, y_train)
+        toc = time.perf_counter_ns()
+
+        train_acc = lr_.score(X_train, y_train)
+        test_acc = lr_.score(X_test, y_test)
+        print(
+            "train acc: %0.3f, test acc: %0.3f, time: %.2fms"
+            % (train_acc, test_acc, (toc - tic) / 1_000_000)
+        )
+        return 0
+
+
+sklearn = XP("numpy", SK_LogisticRegression)
+numpy = XP("numpy", XP_LogisticRegression)
+cupy = XP("cupy", XP_LogisticRegression)
+cunumeric = XP("cunumeric", XP_LogisticRegression)
+
+
+b = perfplot.bench(
+    labels=["sklearn", "NumPy", "CuPy", "cunumeric"],
+    setup=[sklearn.setup, numpy.setup, cupy.setup, cunumeric.setup],
+    kernels=[sklearn.time, numpy.time, cupy.time, cunumeric.time],
+    n_range=[10_000, 100_000, 1_000_000],
+    title="Logistic regression comparison")
+
+
+b.save("out.png")
```

Comment:
```suggestion
    xlabel="Number of samples",
    title="Logistic regression comparison")


b.plot()
plt.ylim([None, 2e2])
plt.savefig("out.png", bbox_inches="tight", transparent=False)
plt.close()
```

To make the y-scale consistent between different plots (GPU vs no GPU)